inductiva/benchmarks/methods.py

```diff
@@ -0,0 +1,370 @@
+                               overwrite=False,
+                               **input_args)
+    return target_dir
+
+
+def run(name: str,
+        input_files: types.Path,
+        replicas: int,
+        machines: List[Dict[str, Any]],
+        simulator: Simulator,
+        append: bool = False,
+        machine_class: BaseMachineGroup = None,
+        input_args: Dict[str, Union[Callable, Any]] = None,
+        machine_args: Dict[str, Union[Callable, Any]] = None,
+        simulator_args: Dict[str, Union[Callable, Any]] = None):
+    """Runs a benchmark.
+    This method creates a project and runs a benchmark using the provided
+    parameters. It checks if a benchmark with the given name already exists. If
+    it does, the user is prompted to either append tasks to the existing
+    benchmark or cancel the operation. The method waits for quotas to be
+    available before starting the necessary resources. If the benchmark is
+    already completed, no further action is taken. If the benchmark is
+    incomplete, only the remaining tasks are executed.
+
+    input_args, machine_args and simulator_args can be dictionaries where the
+    values can be callables that will be called with the resource object as
+    parameter (or str machine_type when it comes to machine_args). This is
+    useful when you want to calculate the values based on the machine_group or
+    machine_type.
+    Example:
+        def data_disk_gb(name):
+            #makes the data disk size relative to the number of vcpus
+            vcpu = int(name.split("-")[2])
+            return vcpu+10
+        machine_args={"data_disk_gb": data_disk_gb}
+        
+    Args:
+            name (str): Name of the benchmark.
+            input_files (types.Path): Path to the input files.
+            replicas (int): Number of replicas to run on each resource.
+            machines (List[Dict[str, Any]]): List of machines to run the
+                simulation on.
+                The machine in this case is a dict with a lot of information.
+                It is mandatory to have the machine_type key and value.
+                Usually used in conjunction with inductiva.resources.query
+            simulator (Simulator): Simulator to run.
+            append (bool): If True, appends the tasks to a benchmark regardless 
+            if that benchmark is new or not. If False, will prompt the user if
+            the benchmark already exists.
+            machine_class (BaseMachineGroup): Class to use to create the machine
+                group. If not given, MachineGroup will be used.
+            input_args (Dict[str, Union[Callable, Any]]): Arguments for the
+                input files.
+            machine_args (Dict[str, Union[Callable, Any]]): Arguments for the
+                machines.
+            simulator_args (Dict[str, Union[Callable, Any]]): Arguments for the
+                simulator.
+    """
+    project = Project(name, append=True)
+    project.open()
```

Comment:
let's wrap the entire body of this function inside a context manager to ensure that the project is always closed:

```python
with Project(name, append=True) as project:
    _run(arg1, arg2, arg3, ...) <-- do work here
```